src/libYARP_OS/src/RateThread.cpp

```diff
@@ -299,14 +320,45 @@ RateThread::~RateThread()
     }
 }
 
+bool RateThread::checkRequiredPeriod(int &period)
```

Comment:
My two cents here: since the actual parameter is an `int` and it is not modified by the function, we should simply pass it by value with a `const int`.